File: problems/algorithms/object-products/ruby/code/spec/products_qtd.spec.rb
require_relative "../src/products_qtd"


products = [
  {
    id: 1,
    preco: 10.0,
    qtd: 2,
  },
  {
    id: 2,
    preco: 10.0,
    qtd: 2,
  },
  {
    id: 3,
    preco: 10.0,
    qtd: 2,
  },
  {
    id: 4,
    preco: 10.0,
    qtd: 0,
  },
]


describe Products do
  describe "Cart Tool" do
    context "filtering greater than zero" do
      it do
        expect(Products.gt_zero(products)).to eql([
          {
            id: 1,
            preco: 10.0,
            qtd: 2,
          },
          {
            id: 2,
            preco: 10.0,
            qtd: 2,
          },
          {
            id: 3,
            preco: 10.0,
            qtd: 2,
          },
        ])
      end
    end

    context "calculating subtotal" do
      it do
        expect(Products.sub_total(products)).to eql([
          {
            id: 1,
            subtotal: 20.0,
          },
          {
            id: 2,
            subtotal: 20.0,
          },
          {
            id: 3,
            subtotal: 20.0,
          },
          {
            id: 4,
            subtotal: 0.0,
          },
        ])
      end
    end

    context "calculating total" do
      it do
        expect(Products.total(products)).to eql(60.0)
      end
    end
  end
end
 

File: src/pages/exercises/array-products-qtd/_codes/php/response/spec/products-qtd.spec.php
<?php
require __DIR__ . '/../src/products-qtd.php';

describe('Cart Tool', function () {

  it('filtering greater than zero', function () {
    $products = [[1, 10.0, 2], [2, 10.0, 2], [3, 10.0, 2], [4, 10.0, 0]];
    expect(gtZero($products))->toEqual([[1, 10.0, 2], [2, 10.0, 2], [3, 10.0, 2]]);
  });

  it('calculating subtotal', function () {
    $products = [[1, 10.0, 2], [2, 10.0, 2], [3, 10.0, 2], [4, 10.0, 0]];
    expect(subTotal($products))->toEqual([[1, 20.0], [2, 20.0], [3, 20.0], [4, 0.0]]);
  });

  it('calculating total', function () {
    $products = [[1, 10.0, 2], [2, 10.0, 2], [3, 10.0, 2], [4, 10.0, 0]];
    expect(total($products))->toBe(60.0);
  });
});
